CMakePresets.json

```diff
@@ -2,83 +2,140 @@
    "version" : 3,
    "configurePresets": [
       {
-         "name" : "base",
-         "displayName" : "Base Configuration",
+         "name" : "base-fp32",
+         "displayName" : "Base FP32 Configuration",
          "warnings": {
             "deprecated": true
          },
          "cacheVariables": {
-            "CMAKE_BUILD_TYPE"        : "RelWithDebInfo",
-            "TRACCC_BUILD_TESTING"    : "TRUE",
-            "TRACCC_BUILD_EXAMPLES"   : "TRUE",
-            "TRACCC_FAIL_ON_WARNINGS" : "TRUE",
-            "TRACCC_USE_ROOT"         : "FALSE"
+            "CMAKE_BUILD_TYPE"         : "RelWithDebInfo",
+            "TRACCC_BUILD_TESTING"     : "TRUE",
+            "TRACCC_BUILD_EXAMPLES"    : "TRUE",
+            "TRACCC_BUILD_BENCHMARKS"  : "TRUE",
+            "TRACCC_FAIL_ON_WARNINGS"  : "TRUE",
+            "TRACCC_USE_ROOT"          : "FALSE",
+            "TRACCC_ALGEBRA_PLUGINS"   : "ARRAY",
+            "TRACCC_CUSTOM_SCALARTYPE" : "float",
+            "DETRAY_CUSTOM_SCALARTYPE" : "float"
          }
       },
       {
-         "name" : "cuda",
-         "displayName" : "CUDA Code Development",
-         "inherits": ["base"],
+         "name" : "base-fp64",
+         "displayName" : "Base FP64 Configuration",
+         "warnings": {
+            "deprecated": true
+         },
+         "cacheVariables": {
+            "CMAKE_BUILD_TYPE"         : "RelWithDebInfo",
+            "TRACCC_BUILD_TESTING"     : "TRUE",
+            "TRACCC_BUILD_EXAMPLES"    : "TRUE",
+            "TRACCC_BUILD_BENCHMARKS"  : "TRUE",
+            "TRACCC_FAIL_ON_WARNINGS"  : "TRUE",
+            "TRACCC_USE_ROOT"          : "FALSE",
+            "TRACCC_ALGEBRA_PLUGINS"   : "ARRAY",
+            "TRACCC_CUSTOM_SCALARTYPE" : "double",
+            "DETRAY_CUSTOM_SCALARTYPE" : "double"
+         }
+      },
+      {
+         "name" : "cuda-fp32",
+         "displayName" : "CUDA FP32 Code Development",
+         "inherits": ["base-fp32"],
          "cacheVariables": {
             "TRACCC_BUILD_CUDA" : "TRUE",
             "VECMEM_BUILD_CUDA_LIBRARY" : "TRUE"
          }
       },
       {
-         "name" : "sycl",
-         "displayName" : "SYCL Code Development",
-         "inherits": ["base"],
+         "name" : "cuda-fp64",
+         "displayName" : "CUDA FP64 Code Development",
+         "inherits": ["base-fp64"],
+         "cacheVariables": {
+            "TRACCC_BUILD_CUDA" : "TRUE",
+            "VECMEM_BUILD_CUDA_LIBRARY" : "TRUE"
```

Comment:
Is `VECMEM_BUILD_CUDA_LIBRARY=TRUE` (or `VECMEM_BUILD_SYCL_LIBRARY=TRUE`) necessary?

Reply:
I thought TRACCC_BUILD_CUDA=true handles other flags well

Reply:
Ahh, that's what you meant... 🤦

Technically, vecmem does auto-detect what it is likely to be able to build. When building the project without presets, one can expect that as long as CUDA is in one's environment, vecmem will automatically build `vecmem::cuda`. But that's not because `TRACCC_BUILD_CUDA` would turn on `VECMEM_BUILD_CUDA_LIBRARY`. But instead because vecmem by itself turns on that flag as long as CUDA is available.

In this way, if CUDA is available in the build environment, even with `--preset base-fp32` the `vecmem::cuda` library will be built.

The reason I ask vecmem explicitly to turn on the build of its CUDA library is to get a slightly clearer error message when things go wrong. If vecmem is told explicitly to build its CUDA code, but for some reason it can't, it will give a relatively clear error message about it. On the other hand if vecmem unexpectedly is unable to build CUDA code, we might end up with an error message in the traccc code that says that `vecmem::cuda` was never defined. At which point users might not understand right away that vecmem's language detection was not recognizing CUDA in the environment.

Long story short: Yes, we should set these things explicitly in the presets. 😉

Reply:
OK